libraries/Microsoft.Bot.StreamingExtensions/Payloads/HeaderSerializer.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using Microsoft.Bot.StreamingExtensions.Transport;
+
+namespace Microsoft.Bot.StreamingExtensions.Payloads
+{
+    internal static class HeaderSerializer
+    {
+        public const byte Delimiter = (byte)'.';
+        public const byte Terminator = (byte)'\n';
+        public const byte End = (byte)'1';
+        public const byte NotEnd = (byte)'0';
+
+        public const int TypeOffset = 0;
+        public const int TypeDelimiterOffset = 1;
+        public const int LengthOffset = 2;
+        public const int LengthLength = 6;
+        public const int LengthDelimeterOffset = 8;
+        public const int IdOffset = 9;
+        public const int IdLength = 36;
+        public const int IdDelimeterOffset = 45;
+        public const int EndOffset = 46;
+        public const int TerminatorOffset = 47;
+
+        public static int Serialize(Header header, byte[] buffer, int offset)
+        {
+            int start = offset;
+
+            buffer[TypeOffset] = (byte)header.Type;
+            buffer[TypeDelimiterOffset] = Delimiter;
+
+            var lengthStr = header.PayloadLength.ToString("D6", CultureInfo.InvariantCulture);
+            Encoding.ASCII.GetBytes(lengthStr, 0, lengthStr.Length, buffer, LengthOffset);
+            buffer[LengthDelimeterOffset] = Delimiter;
```

Comment:
Please include a comment here as to why you're using the ASCII encoding. Many folks won't know. 

I don't understand. :)
[https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html)

Reply:
It's not exactly HTTP, it's part of the bot framework streaming extensions protocol document.

Reply:
I'll add a link to the document once it has a permanent home and is cleared to be public.